framework/src/org/checkerframework/framework/type/visitor/AnnotatedTypeScanner.java

```diff
@@ -129,7 +129,8 @@ public R visitDeclared(AnnotatedDeclaredType type, P p) {
             return visitedNodes.get(type);
         }
         visitedNodes.put(type, null);
-        R r = scan(type.getTypeArguments(), p);
+        R r = scan(type.getEnclosingType(), p);
```

Comment:
I think checker-framework-inference is failing because of this change.

Reply:
This isn't the only place that might pass a null to `scan`.  For example, the return type of an executable could be null.   That's why `scan` checks for null.

Reply:
Wouldn't the return type be the void type? Why didn't CFI fail before this change then?

Reply:
Ok, I looked at the right `scan` now. I think it would be better to never scan null and we should eventually remove all those null checks in `scan`.

Reply:
Right, not the return type, the receiver type. There wasn't a NPE before because `ConstantToVariable` overrides `visitExecutable`  https://github.com/typetools/checker-framework-inference/blob/master/src/checkers/inference/util/ConstantToVariableAnnotator.java#L39

Reply:
I documented that `scan` might scan null (and add more documentation to AnnotatedTypeScanner) in  #1514 .